src/_P144_Vindriktning.ino

```diff
@@ -0,0 +1,376 @@
+    {
+      // Set a default config here, which will be called when a plugin is assigned to a task.
+      success = true;
+      break;
+    }
+
+    case PLUGIN_WEBFORM_LOAD:
+    {
+      // this case defines what should be displayed on the web form, when this plugin is selected
+      // The user's selection will be stored in
+      // PCONFIG(x) (custom configuration)
+
+      // Make sure not to append data to the string variable in this PLUGIN_WEBFORM_LOAD call.
+      // This has changed, so now use the appropriate functions to write directly to the Streaming
+      // web_server. This takes much less memory and is faster.
+      // There will be an error in the web interface if something is added to the "string" variable.
+
+      // Use any of the following (defined at web_server.ino):
+      // addFormNote(F("not editable text added here"));
+      // To add some html, which cannot be done in the existing functions, add it in the following way:
+      /// addRowLabel(F("Analog Pin"));
+
+      // For strings, always use the F() macro, which stores the string in flash, not in memory.
+
+      // String dropdown[5] = { F("option1"), F("option2"), F("option3"), F("option4")};
+      // addFormSelector(string, F("drop-down menu"), F("plugin_xxx_displtype"), 4, dropdown, nullptr, PCONFIG(0));
+
+      // number selection (min-value - max-value)
+      /// addFormNumericBox(string, F("description"), F("plugin_144_description"), PCONFIG(1), min - value, max - value);
+
+      // after the form has been loaded, set success and break
+      //serialHelper_serialconfig_webformLoad(event, true);
+      success = true;
+      break;
+    }
+
+    case PLUGIN_WEBFORM_SAVE:
+    {
+      // this case defines the code to be executed when the form is submitted
+      // the plugin settings should be saved to PCONFIG(x)
+      // ping configuration should be read from CONFIG_PIN1 and stored
+      //serialHelper_webformSave(event);
+
+      // after the form has been saved successfuly, set success and break
+      success = true;
+      break;
+    }
+    case PLUGIN_INIT:
+    {
+      // this case defines code to be executed when the plugin is initialised
+      int8_t rxPin = serialHelper_getRxPin(event);
+      int8_t txPin = serialHelper_getTxPin(event);
+      P144_easySerial = new (std::nothrow) ESPeasySerial(serialHelper_getSerialType(event), rxPin, txPin);
+      if (P144_easySerial != nullptr) 
+      {
+        P144_easySerial->begin(9600);
+        success = true;
+      }
+      String log = F("P144 : Init OK  ESP GPIO-pin RX:");
+      log += rxPin;
+      log += F(" TX:");
+      log += txPin;
+      addLogMove(LOG_LEVEL_INFO, log);
+      break;
+    }
+
+    case PLUGIN_READ:
+    {
+      // code to be executed to read data
+      // It is executed according to the delay configured on the device configuration page, only once
+      UserVar[event->BaseVarIndex] = P144_pm25;
+      String log = F("P144 : READ ");
```

Comment:
Logging should best be wrapped in a condition, to avoid building a String that's possibly never going to be used, like this:
```cpp
if (loglevelActiveFor(LOG_LEVEL_INFO) {
  // Your INFO log
};
```

Reply:
Sorry, did not recognize the intention. Will be updated.